packages/block-editor/src/utils/block-variation-transforms.js

```diff
@@ -24,7 +19,9 @@ export const __experimentalGetMatchingVariation = (
 	if ( ! variations || ! blockAttributes ) return;
 	const matches = variations.filter( ( { attributes } ) => {
 		if ( ! attributes || ! Object.keys( attributes ).length ) return false;
-		return isMatch( blockAttributes, attributes );
+		return Object.entries( attributes ).every(
+			( [ key, value ] ) => blockAttributes[ key ] === value
```

Comment:
This is used currently in Query Loop placeholder and `isMatch` from lodash does deep comparison. We are changing the comparison here and returns different results in case of `object` attributes. Check the recording: first is at trunk and then I reload with your branch.

https://user-images.githubusercontent.com/16275880/178248451-3cab5b65-8a8c-42d9-9fa0-7eb8d4621d85.mov

Reply:
Thanks for adding the tests!